### PR TITLE
Reader: Delete local reader posts in db migration (duplicate posts issue - part 3)

### PR DESCRIPTION
**Pull request by ashiagr**
Issue #12938 - Part 3

Part 1: #14813 (discusses the followed-sites duplicate post issue, endpoint fix D63092-code)
Part 2: #14879 (fixes discover/ blog posts duplicates)

Before deploying D63092-code, while thinking about regressions, it was noticed that in the app upgrade scenario, [posts with invalid pseudo ids will still exist in the DB](https://cloudup.com/ivBtGyPPR-y) and can re-create duplicates when endpoint fix is deployed which will provide correct `pseudo_id` for Jetpack posts having shadow `wpcom` posts.

This PR 
- deletes (`Followed-Sites` + blog) posts in a DB migration. It also updates `date_updated` for the `Followed-Sites` tag in `tbl_tag` so that these posts for the tag are re-fetched [here](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java#L2090) after this migration. - dc862f3
- updates bookmarked posts with valid pseudo ids when `Followed-Sites` posts are re-fetched after the migration. - 9812c27

### To test

I followed the below steps to test it:

1. Installed a previous version, followed steps in #14813, bookmarked a duplicate post so that posts with invalid pseudo ids are created in the DB. 
2. Pulled the patch D63092-code in the sandbox, sandboxed the device using Charles, installed the code from this branch without deleting the already installed app (so that migration gets triggered) -> launched the app -> went to Reader and repeated duplicate posts generation steps.
3. Noticed that all posts had correct pseudo ids and noticed no duplicate posts for a blog.

    [Posts with invalid pseudo ids in the DB before app upgrade](https://cloudup.com/ivBtGyPPR-y).
    [Posts with valid pseudo ids in the DB after app upgrade](https://cloudup.com/inels8Q_BA6).

### Review Instructions:
- Let me know if you have any issues testing on a sandbox. We can pair and I can show it on my sandbox.

### Merge Instructions:
Wait for merge till D63092-code is deployed.

## Regression Notes
1. Potential unintended areas of impact 🟢 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
N/A (changes in DB migration)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ParaskP7 — COMMENTED** [review]
👋 @ashiagr !

I have reviewed this PR, everything LGTM, good job! 🌟

However, before me proceeding with approving this PR could we pair on it for few minutes? I would like to:
1) Test it together with you on the sandbox.
2) Ask you a couple of question about my assumptions on what is happening and why.

🙏

File: WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java
Comment (ParaskP7) [inline WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java]: Minor (🔍): Consider renaming that to `bookmarkedPostId` instead of having it a plural, which might suggest that this is somehow a list.

PS: It is already a singular on the already existing `isBookmarkedPost(...)` method.

**Review by ParaskP7 — APPROVED** [review]
👋 @ashiagr !

After pairing on that, and as discussed, I am now approving this PR since everything work as expected. Once more good job! 🌟 

Feel free to merge this PR when you are ready.

File: WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java
Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/datasets/ReaderPostTable.java]: Fixed in 0455b05

**Comment by ashiagr** [comment]
Thank you @ParaskP7! Really appreciate your time to discuss this PR.
Addressed the review comment and added a release note for user-facing change. 

Going to deploy the D63092-code now.

**Pull request merged**
